### PR TITLE
'patch article votes completed with error checks'

### PR DESCRIPTION
**Pull request by WelshSquirrel**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -9,77 +9,150 @@ beforeEach(() => {
...
+            .expect(404)
+            .then(({
+                body
+            }) => {
+                expect(body).toEqual({
+                    msg: 'article ID not found'
+                })
+            })
+    })
+})
```
Comment (popatre) [inline __tests__/app.test.js:201]: Some missing tests to add:

- [x] 400 invalid id
- [x]  200/400 inc_vote key missing - either respond with an error or make no change.

**Review by popatre — CHANGES_REQUESTED** [review]
Just needs a few more tests

**Comment by WelshSquirrel** [comment]
I have commented out a section of tests that crossed over from my next branch.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -33,50 +33,68 @@ describe('GET /api/topics', () => {
...
-            .then(({
-                body
-            }) => {
-                expect(body).toEqual({
-                    msg: 'article ID not found'
-                })
-            })
-    })
-})
+// describe('GET /api/articles/:article_id', () => {
```
Comment (popatre) [inline __tests__/app.test.js:36]: Ideally these shouldn't even exist on this branch, as this branch is for a different feature. 

I'd make sure you're branching from main each time, then all the branches are independent, and all feed into main

File: models/models.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ exports.gatherTopics = () => {
 }
 
 exports.gatherArticles = (id) => {
-    return db.query('SELECT * FROM articles WHERE article_id = $1;', [id])
+    return db.query('SELECT * FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id WHERE article_id = $1;', [id])
```
Comment (popatre) [inline models/models.js:10]: I presume this is the next ticket, so should be it's own branch.

**Review by popatre — APPROVED** [review]

**Pull request merged**